src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/RoundTripIslSpec.groovy

```diff
@@ -0,0 +1,287 @@
+        }
+
+        then: "The round trip ISL is still DISCOVERED (the system uses round trip latency for ISL alive confirmation)"
+        sleep((discoveryTimeout + WAIT_OFFSET / 2) * 1000 as long)
+        northbound.getLink(roundTripIsl).state == IslChangeType.DISCOVERED
+
+        and: "Round trip status for forward direction is not available and ACTIVE in reverse direction"
+        !database.getIslRoundTripStatus(roundTripIsl)
+        database.getIslRoundTripStatus(roundTripIsl.reversed) == IslStatus.ACTIVE
+
+        when: "Delete ROUND_TRIP_LATENCY_RULE_COOKIE on the dst switch"
+        northbound.deleteSwitchRules(dstSw.dpId, DeleteRulesAction.REMOVE_ROUND_TRIP_LATENCY)
+        def isRoundTripRuleDeleted = true
+        Wrappers.wait(RULES_DELETION_TIME) {
+            assert northbound.validateSwitch(dstSw.dpId).rules.missing.size() == 1
+        }
+
+        then: "The round trip latecny ISL is FAILED"
+        sleep((discoveryTimeout + WAIT_OFFSET / 2) * 1000 as long)
+        northbound.getLink(roundTripIsl).state == IslChangeType.FAILED
+
+        and: "Round trip status is not available for the given ISL in both directions"
+        [roundTripIsl, roundTripIsl.reversed].each { isl ->
+            assert !database.getIslRoundTripStatus(isl)
+        }
+
+        when: "Install ROUND_TRIP_LATENCY_RULE_COOKIE on the dst switch"
+        northbound.installSwitchRules(dstSw.dpId, InstallRulesAction.INSTALL_ROUND_TRIP_LATENCY)
+        isRoundTripRuleDeleted = false
+        Wrappers.wait(RULES_INSTALLATION_TIME) {
+            assert northbound.validateSwitch(dstSw.dpId).rules.missing.empty
+        }
+
+        then: "The round trip ISL is still FAILED"
+        //ISL will be in the FAILED state until gets any discovery packet
+        sleep((discoveryInterval + WAIT_OFFSET / 2) * 1000 as long)
+        northbound.getLink(roundTripIsl).state == IslChangeType.FAILED
+
+        when: "Delete ROUND_TRIP_LATENCY_RULE_COOKIE on the dst switch again"
+        northbound.deleteSwitchRules(dstSw.dpId, DeleteRulesAction.REMOVE_ROUND_TRIP_LATENCY)
+        isRoundTripRuleDeleted = true
+        Wrappers.wait(RULES_DELETION_TIME) {
+            assert northbound.validateSwitch(dstSw.dpId).rules.missing.size() == 1
+        }
+
+        and: "Activate the src switches"
+        lockKeeper.reviveSwitch(srcSwToDeactivate)
+        Wrappers.wait(discoveryInterval + WAIT_OFFSET) {
+            assert northbound.getSwitch(srcSwToDeactivate.dpId).state == SwitchChangeType.ACTIVATED
+            assert northbound.getAllLinks().findAll {
+                it.state == IslChangeType.DISCOVERED
+            }.size() == topology.islsForActiveSwitches.size() * 2
+        }
+        isSrcSwDeactivated = false
+
+        then: "Round trip isl is DISCOVERED again"
+        northbound.getLink(roundTripIsl).state == IslChangeType.DISCOVERED
+
+        and: "Round trip status is available again for the given ISL in both directions"
```

Comment:
Why do you expect it to be active in both directions if you removed the RTL rule on L194 and never installed it back?